tests/testthat/test_agg.R

```diff
@@ -0,0 +1,346 @@
+library(data.table)
+library(testthat)
+
+# Aggregate categorical variable with varying interval id cols ------------
+
+# Inputs:
+# - males: 5-calendar-year interval and 5-year age groups
+# - females: 1-calendar-year interval and 1-year age groups
+# Output:
+# - all sexes combined: 5 calendar-year interval and 5-year age groups
+
+sex_mapping <- data.table(parent = "both", child = c("female", "male"))
+id_cols <- c("year_start", "year_end", "sex", "age_start", "age_end")
+value_cols <- c("value")
+
+# set up test input data.table
+input_dt_male <- CJ(
+  year_start = 2005, year_end = 2010,
+  sex = "male",
+  age_start = seq(0, 95, 5),
+  value = 25
+)
+input_dt_male[age_start == 95, value := 5]
+
+input_dt_female <- CJ(
+  year_start = 2005:2009,
+  sex = "female",
+  age_start = seq(0, 95, 1),
+  value = 1
+)
+gen_end(
+  input_dt_female,
+  id_cols = setdiff(id_cols, c("year_end", "age_end")),
+  col_stem = "year",
+  right_most_endpoint = 2010
+)
+
+input_dt <- rbind(input_dt_male, input_dt_female)
+gen_end(input_dt, id_cols = setdiff(id_cols, "age_end"), col_stem = "age")
+setkeyv(input_dt, id_cols)
+
+# set up expected output data.table
+expected_dt <- CJ(
+  year_start = 2005, year_end = 2010,
+  sex = "both",
+  age_start = seq(0, 95, 5),
+  value = 50
+)
+expected_dt[age_start == 95, value := 10]
+gen_end(expected_dt, id_cols = setdiff(id_cols, "age_end"), col_stem = "age")
+setkeyv(expected_dt, id_cols)
+
+description <- "aggregating a categorical variable with varying interval id cols
+works"
+test_that(description, {
```

Comment:
Why did you decide to stop using `testthat::` before these functions?

Reply:
This wasn't intentional, was copying over the tests previously used in this package.

But actually I think I like this better for testing scripts. Similar to how we expect developers to be familiar with data.table, for testing scripts maybe `testthat` is also okay to not use `::` notation .

What do you think?